File: sample-app-sticker-camera/src/main/java/com/allever/sticker/ui/MainActivity.java
package com.allever.sticker.ui;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.os.Process;
import android.view.View;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.annotation.Nullable;

import com.isseiaoki.simplecropview.util.Utils;
import com.allever.sticker.ControllerEnum;
import org.xm.sticker.camera.R;

import app.allever.android.lib.core.base.AbstractActivity;


/**
 *
 * @author Allever
 * @date 2018/1/1
 */

public class MainActivity extends AbstractActivity implements View.OnClickListener{
    private static final String TAG = "MainActivity";

    private static final int REQUEST_PICK_IMAGE = 1001;
    private static final int REQUEST_SAF_PICK_IMAGE = 1002;
    private static final int RESULD_CODE_TAKE_PHOTO = 1004;

    private ImageView btn_pick_img;
    private ImageView btn_camera;
    private ImageView btn_store;

    private Uri mImageUri;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.sc_activity_main);

        initView();
    }

    private void initView(){
        btn_pick_img = (ImageView)findViewById(R.id.id_main_btn_pick_img);
        btn_pick_img.setOnClickListener(this);
        btn_camera = (ImageView)findViewById(R.id.id_main_btn_camera);
        btn_camera.setOnClickListener(this);
        btn_store = (ImageView)findViewById(R.id.id_main_btn_store);
        btn_store.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        int id = view.getId();
        if (id == R.id.id_main_btn_pick_img) {//如果有存储权限，则打开相册选择图片，否则申请存储权限
            ControllerEnum.chooseImageFromGallery(this, REQUEST_PICK_IMAGE);
        } else if (id == R.id.id_main_btn_camera) {//如果有相机权限，则打开相机，否则申请相机权限
            mImageUri = ControllerEnum.openCamera(this, RESULD_CODE_TAKE_PHOTO);
        } else if (id == R.id.id_main_btn_store) {
            StoreActivity.startSelf(this);
        }
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent result) {
        super.onActivityResult(requestCode, resultCode, result);
        if (resultCode == Activity.RESULT_OK) {
            switch (requestCode) {
                case REQUEST_PICK_IMAGE:
                    CropActivity.startSelf(this,result.getData());
                    break;
                case REQUEST_SAF_PICK_IMAGE:
                    CropActivity.startSelf(this, Utils.ensureUriPermission(MainActivity.this, result));
                    break;
                case RESULD_CODE_TAKE_PHOTO:
                    CropActivity.startSelf(this,mImageUri);
                    break;
                default:
                    break;
            }
        }
    }

    private long mPrevClickBackTime = -1;
    @SuppressLint("MissingSuperCall")
    @Override
    public void onBackPressed() {
        long currentTime = System.currentTimeMillis();
        if (mPrevClickBackTime == -1 || currentTime - mPrevClickBackTime > 3000) {
            mPrevClickBackTime = currentTime;
            Toast.makeText(this, "Press again to exit",
                    Toast.LENGTH_LONG).show();
            return;
        }
        Process.killProcess(Process.myPid());
    }
}

File: sample-app-spy-camera/src/main/java/com/allever/stealthcamera/ui/MainActivity.kt
package com.allever.stealthcamera.ui

import android.Manifest
import android.app.AlertDialog
import android.content.DialogInterface
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import app.allever.android.lib.core.base.AbstractActivity
import app.allever.android.lib.core.ext.toast

import com.allever.stealthcamera.FloatWindowService
import org.xm.stealth.camera.R
import com.allever.stealthcamera.function.permission.FloatWindowManager
import com.allever.stealthcamera.function.permission.rom.RomUtils
import com.allever.stealthcamera.function.permission.rom.VivoUtils
import com.allever.stealthcamera.utils.CameraUtil

class MainActivity : AbstractActivity() {

    private var mIvCam: ImageView? = null
    private var mIvSetting: ImageView? = null
    private var mIvPic: ImageView? = null
    private var mIvGenCam: ImageView? = null

    private var mPrevClickBackTime: Long = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initData()
        initView()
    }

    private fun initData() {}

    private fun initView() {
        mIvCam = findViewById(R.id.id_main_iv_camera)
        mIvSetting = findViewById(R.id.id_main_iv_settings)
        mIvPic = findViewById(R.id.id_main_iv_pictures)
        mIvGenCam = findViewById(R.id.id_main_iv_general_camera)

        if (FloatWindowService.mService == null) {
            mIvCam!!.setImageResource(R.drawable.ic_camera_off)
        } else {
            mIvCam!!.setImageResource(R.drawable.ic_camera_on)
        }

        mIvCam?.setOnClickListener {
            if (FloatWindowManager.applyOrShowFloatWindow(this@MainActivity)) {
                val floatIntent = Intent(this@MainActivity, FloatWindowService::class.java)
                if (FloatWindowService.mService == null) {
                    startService(floatIntent)
                    mIvCam?.setImageResource(R.drawable.ic_camera_on)
                } else {
                    stopService(floatIntent)
                    mIvCam?.setImageResource(R.drawable.ic_camera_off)
                }
            } else {
                showSettingDialog()
            }
        }

        mIvSetting?.setOnClickListener {
            //设置界面
            val intent = Intent(this@MainActivity, SettingActivity::class.java)
            startActivity(intent)
        }

        mIvPic?.setOnClickListener {
            val intent = Intent(this@MainActivity, PictureActivity::class.java)
            startActivity(intent)
        }

        mIvGenCam?.setOnClickListener(View.OnClickListener {
            if (!CameraUtil.checkCameraHardware(this@MainActivity)) {
                return@OnClickListener
            }
            if (FloatWindowService.mService != null) {
                //停止预览
                val floatIntent = Intent(this@MainActivity, FloatWindowService::class.java)
                stopService(floatIntent)
                mIvCam?.setImageResource(R.drawable.ic_camera_off)
            }

            val intent = Intent(this@MainActivity, CameraActivity::class.java)
            startActivity(intent)
        })


    }

    private fun showSettingDialog() {
        AlertDialog.Builder(this)
                .setTitle(R.string.string_dialog_title)
                .setMessage(R.string.string_dialog_message)
                .setPositiveButton(R.string.string_dialog_setting_button) { dialogInterface, i ->
                    if (RomUtils.checkIsVivoRom()) {
                        VivoUtils.applyOppoPermission(this@MainActivity)
                    } else {
                        FloatWindowManager.applyPermission(this@MainActivity)
                    }
                }
                .setNegativeButton(R.string.string_dialog_cancel_button) { dialogInterface, i -> }
                .show()
    }

    override fun onBackPressed() {
        val currentTime = System.currentTimeMillis()
        if (mPrevClickBackTime == -1L || currentTime - mPrevClickBackTime > 3000) {
            mPrevClickBackTime = currentTime
            toast("Press again to exit")
            return
        }
        super.onBackPressed()
    }
}
